### PR TITLE
NL-39 CRUD Functions for Goals

### LINKED ISSUE #39: Homepage Backend
## Name and Email
  - Name: Yen-Yun Wu and Rickesh Khilnani
  - Email: [yew005@ucsd.edu](mailto:yew005@ucsd.edu) and [rikhilna@ucsd.edu](mailto:rikhilna@ucsd.edu)

## Complexity of Issue (Scale from 1-5)
  - 4
  
## Issue Description
  - Create the backend CRUD functionality for the Homepage

### LINKED ISSUE #46: CRUD functions for Goals
## Name and Email
  - Name: Dustin
  - Email: d6lin@ucsd.edu

## Complexity of Issue (Scale from 1-5)
  - 3
  
## Issue Description
  - update the existing CRUD functions in the backendinit file to use firebase, try to have the same behavior

### PR DESCRIPTION
**Pull request by Rick10101221**
Closes #46 #39

### CODE REVIEW

**Comment by Rick10101221** [comment]
Need to add comments

File: .eslintrc.json [hunk 1]
```diff
@@ -19,12 +19,8 @@
...
     "overrides": [
       {
-        "files": [ 
-            "FirebaseInit.js",
-            "firebase-app.min.js",
-            "firebase-auth.min.js",
-            "firebase-database.min.js",
-            "Login.js"
+        "files": [
+            "*.js"
```
Comment (DustinLin) [inline .eslintrc.json:23]: is this alright? I know @nvanny had some errors relating to this, so maybe it's fine overwriting the rules for all files?

File: source/Login/Login.js [hunk 2]
```diff
@@ -47,9 +47,9 @@ function signIn() {
         signInWithEmailAndPassword(auth, userEmail, password)
             // eslint-disable-next-line no-unused-vars
             .then((userCredential) => {
-                // TODO
+                // TODO:
                 alert('Successfully signed in!');
-                window.location.replace('../Index/Index.html');
+                window.location.replace('../Index/TEMP-REMOVE/test.html');
```
Comment (DustinLin) [inline source/Login/Login.js]: Not sure what this is, will it be removed?

File: source/YearlyOverview/Yearly.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { db } from '../Backend/FirebaseInit.js';
+import {
+    getUserID,
+    // setObjAtDBPath,
+    // getCurrentDate,
+} from '../../General/GlobalUtility.js';
```
Comment (DustinLin) [inline source/YearlyOverview/Yearly.js]: This is a good file that everyone on the team should know

**Review by DustinLin — CHANGES_REQUESTED** [review]
refer to comments

File: .eslintrc.json (hunk 1 above)
Comment (Rick10101221) [reply .eslintrc.json:23]: This should be alright. It pretty much just makes eslint parse all _.js_ files as modules. If @nvanny still experiences errors, I'd ask him to post it on slack, and if it's unsolvable, then we can instead list specific directory paths under "files" here

File: source/YearlyOverview/Yearly.js (hunk 3 above)
Comment (Rick10101221) [reply source/YearlyOverview/Yearly.js]: Is it alright if I remove this file? `Yearly.js` was meant to be a refactored version of YearlyJS.js

File: source/Login/Login.js (hunk 2 above)
Comment (Rick10101221) [reply source/Login/Login.js]: Let me revert this back to `Index.html` that's my bad.

File: source/YearlyOverview/Yearly.js (hunk 3 above)
Comment (DustinLin) [reply source/YearlyOverview/Yearly.js]: file seems removed, it was used for testing but now it's functions are in `backendInit.js`

**Review by DustinLin — APPROVED** [review]

**Pull request merged**